src/Microsoft.OpenApi/Services/OpenApiWalker.cs

```diff
@@ -1257,6 +1325,62 @@ private void Walk(string context, Action walk)
             _visitor.Exit();
         }
 
+        /// <summary>
+        /// Adds a segment to the context path to enable pointing to the current location in the document
+        /// </summary>
+        /// <typeparam name="T">The type of the state.</typeparam>
+        /// <param name="context">An identifier for the context.</param>
+        /// <param name="state">The state to pass to the walk action.</param>
+        /// <param name="walk">An action that walks objects within the context.</param>
+        private void WalkItem<T>(string context, T state, Action<OpenApiWalker, T> walk)
+        {
+            _visitor.Enter(context.Replace("/", "~1"));
```

Comment:
this replacement should be its own method instead of being duplicated.
And as we refactor it, we could use conditional compilation so for everything netstandad 2.1 and up, we pass the string comparison